nodebuilder/p2p/p2p.go

```diff
@@ -0,0 +1,224 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	libhost "github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/metrics"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	rcmgr "github.com/libp2p/go-libp2p-resource-manager"
+	basichost "github.com/libp2p/go-libp2p/p2p/host/basic"
+	"github.com/libp2p/go-libp2p/p2p/net/conngater"
+)
+
+// API is a wrapper around Module for the RPC.
+// TODO(@distractedm1nd): These structs need to be autogenerated.
+//
+//nolint:dupl
+type API struct {
+	Info                 func() Info
+	Peers                func() []peer.ID
+	PeerInfo             func(id peer.ID) peer.AddrInfo
+	Connect              func(ctx context.Context, pi peer.AddrInfo) error
+	ClosePeer            func(id peer.ID) error
+	Connectedness        func(id peer.ID) network.Connectedness
+	NATStatus            func() (network.Reachability, error)
+	BlockPeer            func(p peer.ID) error
+	UnblockPeer          func(p peer.ID) error
+	ListBlockedPeers     func() []peer.ID
+	Protect              func(id peer.ID, tag string)
+	Unprotect            func(id peer.ID, tag string) bool
+	IsProtected          func(id peer.ID, tag string) bool
+	BandwidthStats       func() metrics.Stats
+	BandwidthForPeer     func(id peer.ID) metrics.Stats
+	BandwidthForProtocol func(proto protocol.ID) metrics.Stats
+	ResourceState        func() (rcmgr.ResourceManagerStat, error)
+	PubSubPeers          func(topic string) []peer.ID
+}
```

Comment:
Readability nit: let's keep this auxiliary API struct somewhere below of the file, while Module the interface at the top

Reply:
kk